pkg/controller/kieapp/defaults/defaults_test.go

```diff
@@ -7491,3 +7502,26 @@ func TestRhpamKieserverEnvWithoutDecisionsOnlyEnabled(t *testing.T) {
 	result, _ := strconv.ParseBool(getEnvVariable(env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_SERVER_DECISIONS_ONLY"))
 	assert.False(t, result)
 }
+
+func TestCredentialWithCredentialSecret(t *testing.T) {
```

Comment:
please improve this tests and add more test cases
for this test, as the secret was provided together the admin user and password, the secret should be taken and its value used as field ref, you need to create a secret, reconcile and check if the deployments contains it where it is nedded.


additional test cases:
- pass only the credentials and check if the secret is created with the provided values and the plain text credentials removed from kieApp cr and assert if the secret is created with the desired values plus assert the deployments.

Reply:
Done